app.py

```diff
@@ -44,6 +46,13 @@ def interpolate_index(self, **kwargs):
     external_stylesheets=external_stylesheets,
 )
 app.static_folder = "assets"
+if "DEBUG" not in os.environ:
```

Comment:
Features which it's only possible to enable by disabling DEBUG are a bit of a Django anti-pattern in my view and can make debugging and testing unnecessarily difficult. I think the fix here is just to have:
```python
credentials = os.environ.get("BASIC_AUTH_CREDENTIALS")

if "DEBUG" not in os.environ:
    assert (
        credentials
    ), 'You need to `export BASIC_AUTH_CREDENTIALS="username: password"`'

if credentials:
    username, password = [x.strip() for x in credentials.split(":")]
    auth = dash_auth.BasicAuth(app, {username: password})
```
i.e. credentials are optional in DEBUG mode and mandatory otherwise.

Reply:
Yes, agreed!